rules/okta_rules/okta_potentially_stolen_session.py

```diff
@@ -0,0 +1,94 @@
+import json
+from datetime import datetime, timedelta
+from difflib import SequenceMatcher
+
+from panther_base_helpers import deep_get, okta_alert_context
+from panther_oss_helpers import get_string_set, put_string_set, set_key_expiration
+
+FUZZ_RATIO_MIN = 0.95
+PREVIOUS_SESSION = {}
+SESSION_TIMEOUT = 1  # the number of days an Okta session is valid for (configured in Okta)
+
+
+def rule(event):
+    # pylint: disable=global-statement
+    # ensure previous session info is avaialable in the alert_context for investigation
+    global PREVIOUS_SESSION
+
+    session_id = deep_get(
+        event, "authenticationContext", "externalSessionId", default="session_not_found"
+    )
+
+    # Filter only on app access and session start events
```

Comment:
I typically like to not do big if-else blocks and just break early:

```python
if not filter-logic:
  return False
```

Reply:
Ah good call, will adjust